File: typescript/test/model/FakeCatalog.ts
import {Product} from "../../src/model/Product"
import {SupermarketCatalog} from "../../src/model/SupermarketCatalog"

export class FakeCatalog implements SupermarketCatalog {
    private products: {[key: string]: Product} = {};
    private prices: {[key: string]: number} = {};

    public addProduct(product: Product, price: number): void {
        this.products[product.getName()] = product;
        this.prices[product.getName()] = price;
    }

    public getUnitPrice(p: Product): number {
        return this.prices[p.getName()];
    }
}

File: swift/SupermarketReceipt/SupermarketReceiptTests/FakeCatalog.swift
public class FakeCatalog: SupermarketCatalog {
    private var products = [String: Product]()
    private var prices = [String: Double]()

    public func addProduct(product: Product, price: Double) {
        self.products.append(product.name, product)
        self.prices.append(product.name, price)
    }

    public func getUnitPrice(p: Product) -> Double {
        return this.prices.get(p.getName());
    }
}
